pkg/notifications/move_counseled.go

```diff
@@ -91,12 +91,19 @@ func (m MoveCounseled) emails(appCtx appcontext.AppContext) ([]emailContent, err
 		return emails, fmt.Errorf("no email found for service member")
 	}
 
+	var weightRestriction *int64
+	if orders.Entitlement.WeightRestriction != nil {
+		weightRestrictionInt64 := int64(*orders.Entitlement.WeightRestriction)
+		weightRestriction = &weightRestrictionInt64
+	}
```

Comment:
```suggestion
	if orders.Entitlement != nil && orders.Entitlement.WeightRestriction != nil {
		weightRestrictionInt64 := int64(*orders.Entitlement.WeightRestriction)
		weightRestriction = &weightRestrictionInt64
	}
```

Sneaky nil check, since it's nested you gotta check both

Reply:
Eg if you try to look for something equalling nil inside of something that can be nil, it may attempt memory referencing something that doesn't exist in memory

Reply:
added